src/Illuminate/Database/Eloquent/Relations/Relation.php

```diff
@@ -465,6 +465,17 @@ public static function getMorphedModel($alias)
         return static::$morphMap[$alias] ?? null;
     }
 
+    /**
+     * Get the alias associated with a given Model class name.
+     *
+     * @param  string  $model
+     * @return false|int|string
+     */
+    public static function getMorphedAlias($model)
+    {
+        return array_search($model, static::$morphMap);
+    }
```

Comment:
I would do something like this:

```suggestion
    /**
     * Get the alias associated with a given Model class name.
     *
     * @param  string|\Illuminate\Database\Eloquent\Model  $model
     * @return  string
     */
    public static function getMorphedAlias($model)
    {
        if (! is_string($model)) {
            $model = get_class($model);
        }
        
        $morphMap = Relation::morphMap();
        
        if (! empty($morphMap) && in_array($model, $morphMap)) {
            return array_search($model, $morphMap, true);
        }
        
        return $model;
    }
```